draft-wang-ppm-dap-taskprov.md

```diff
@@ -56,69 +60,302 @@ collector). Thus it is up to each deployment of DAP to securely implement its
 own mechanism.
 
 This document describes a mechanism for configuring tasks that may be useful in
-many deployments.
+many deployments. The goal of this mechanism is to add transparency to the
+task provision process, and define a task provision method that does not rely
+on deployment specific leader-helper out-of-band agreement.
+
+At a high level, this extension asks client to include all the task
+configuration parameters it received out-of-band from task author, in the
+extension field of the `Report` it uploads to aggregators. The aggregators will
+create a DAP task upon receiving `Report` with such extension (or `ReportShare`
+in the case of helper.)
+
+By sending task configuration parameters to clients, we add transparency to the
+task that clients participate in. Client can see what privacy parameters has
+been configured for a task (for e.g. `min_batch_size`, or any differential
+privacy parameters if that's the privacy guarantee used).
+
+By defining this mechanism as an extension, we can guarantee that for any
+deployments implementing this extension:
+
+1. Task transparency is enforced since aggregator task must be created using
+the same parameters client sent.
+
+2. No need for out-of-band task orchestration between leader and helpers, any
+leaders and helpers can work on the same DAP task, therefore making adoption of
+DAP easier.
+
+3. A "task" now means the same task ID and same task configuration, including
+parameters for choosing VDAF. Therefore, reports from one task will only be
+aggregated with one VDAF. For the same reason, if a malicious client changes
+the task ID or task configuration, its report will be aggregated in a different
+task, with other poison reports from the same malicious attack. The "good"
+reports will not be polluted.
```

Comment:
I don't think this property (3.) is new to draft-wang-ppm-taskprov. In fact, it ought to hold for any deployment of DAP.

Reply:
agreed. But I think the part that filters out polluted reports is still worthy calling out, let me move it to Implementation and Operational Considerations, I can delete it if you think that's clearer.